### PR TITLE
[somfytahoma] New channel on the gateway to execute scenes

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@octa22 : this is the implementation of action groups (scenes) through a simple channel on the bridge. For compatibility reasons, I did not remove thing stuff already implemented for action groups. Any user will then have the choice, either using the new channel on the bridge or create a thing for each action group he likes as before.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml:15]: I still think a better place for this channel is Gateway, not Bridge. The Bridge thing is responsible for the cloud communication and should stay as is, without any channels. If something aggregates the devices in the scenarios, it is the Gateway.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml
Comment (octa22) [inline bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:397]: Tahoma users knows ```scenarios``` and Tahoma internally uses ```actionsgroups```. Please do not introduce ```scenes``` and please rename the channel to ```scenarios```.

**Review by octa22 — CHANGES_REQUESTED** [review]
I think having the scenarios as channels is a very interesting idea. But one universal channel seems to me to be very far from great user experience. I am missing any documentation in the readme, how the users would trigger the scenarios from a rule for instance? Do they need to know the internal OID of the actiongroup?

**Comment by lolodomo** [comment]
> I think having the scenarios as channels is a very interesting idea. But one universal channel seems to me to be very far from great user experience.

Having to handle one thing for each action group looks far from great to me, just to select a command amongst a list of commands. That's why I introduced this command channel.

> I am missing any documentation in the readme, how the users would trigger the scenarios from a rule for instance? Do they need to know the internal OIDvof the actiongroup?

Good point, I forgot the documentation. And yes the users would have to use the internal OID of the action group in rules.
In UI, the action group label is automatically displayed to the user.

I already did that in other bindings like the hue binding for example, I added a console command to list the scenerios with their idea. I will do that.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml:15]: Ok, why not. I will move the channel. It makes the thing just a little more difficult.

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:397]: I just used scene as I have the feeling that scene is the correct english word for what is a "scenario" in French.

I just found that:: "As nouns the difference between scene and scenario is that scene is the location of an event that attracts attention while scenario is an outline of the plot of a dramatic or literary work. "
So yes, scenario is the right word to use I think (now).

Comment (octa22) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml:397]: good to know it :-) So the Tahoma guys have translated it right

**Comment by lolodomo** [comment]
I will have to wait for the merge of PR #10355

**Review by octa22 — APPROVED** [review]
LGTM

**Review by octa22 — CHANGES_REQUESTED** [review]
Please update the documentation

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/bridge.xml]: Done

File: bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/resources/OH-INF/thing/channels.xml]: Done

**Comment by lolodomo** [comment]
Yes, remain to do to cover all your remarks:

* documentation
* console command

**Comment by lolodomo** [comment]
@octa22 : ready for a final review.

**Comment by lolodomo** [comment]
@octa22 : can you please finally validate ?

**Review by octa22 — APPROVED** [review]
LGTM. Thanks

**Comment by lolodomo** [comment]
@openhab/add-ons-maintainers : please review & merge.

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java
Comment (fwolter) [inline bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:107]: What is the reason for setting the Channel with the updated value manually?

**Review by fwolter — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java
Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:107]: Because the channel state was not set if I did not add this line.
I tried only with MainUI. Could it be a setup problem with my item or a UI bug ?
I will try again, including with Basic UI.

Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:107]: Did you try to set the auto update policy to `recommended`? https://www.openhab.org/docs/developer/bindings/thing-xml.html#auto-update-policies

Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:107]: Why the default policy would not work?

Comment (fwolter) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java:107]: I'm not aware of the details of the auto update algorithm, but I think the default setting expects the Channel to be updated when the Thing is polled the next time. This will never happen in your case, so you can adjust the auto update policy to tell the framework to update the Channel immediately. Then, you can remove the `updateState()` and the model will also tell one the intented behavior of the Channel.

Comment (lolodomo) [reply bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaGatewayHandler.java]: I agree that it is the good approach. Tested and working.

**Comment by lolodomo** [comment]
Sorry, I introduced a bad character in one of my commit in my email address. I believe it explains the DCO error. I tried to follow the instructions to fix it but without success.

**Comment by fwolter** [comment]
Never mind, we can force the DCO to pass.

**Comment by lolodomo** [comment]
@fwolter : build succeeded ;)

**Review by fwolter — APPROVED** [review]

**Pull request merged**
